### ISSUE TITLE
a11y red-flags: focus handling

### ISSUE THREAD

**Issue by candu** [CityofToronto/bdit_flashcrow#687]
**Description**
A couple of issues related to focus visibility, ordering, etc.

**Acceptance Criteria**
- [ ] all elements in focus should be visibly outlined;
- [ ] should not use `tabindex` on non-interactive elements;
- [ ] save focus when opening modals, and restore it when closing them;
- [ ] address issues with autofocus on location search bar:
  - [ ] on first load, skips navigation and doesn't allow user to read that;
  - [ ] in Add Location, doesn't restore focus when closed;

**Comment by candu** [comment]
Closing this now that we have our full a11y feedback - we'll create more specific tasks there.
